src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -511,9 +500,10 @@ function Set-AzConnectedKubernetes {
 
         $options = ""
 
-        if ($DisableAutoUpgrade) {
+        if ($DisableAutoUpgrade -or ($ExistConnectedKubernetes.ArcAgentProfileAgentAutoUpgrade -eq 'Disabled')) {
             $options += " --set systemDefaultValues.azureArcAgents.autoUpdate=false"
```

Comment:
As before, should this be set via ConfigDP and removed here?

Reply:
Yes, nice catch!